pkg/textparse/openmetricsparse.go

```diff
@@ -81,15 +81,17 @@ func (l *openMetricsLexer) Error(es string) {
 // OpenMetrics text exposition format.
 // This is based on the working draft https://docs.google.com/document/u/1/d/1KwV0mAXwwbvvifBvDKH_LU1YjyXE_wxCkHNoCGq1GX0/edit
 type OpenMetricsParser struct {
-	l       *openMetricsLexer
-	series  []byte
-	text    []byte
-	mtype   MetricType
-	val     float64
-	ts      int64
-	hasTS   bool
-	start   int
-	offsets []int
+	l              *openMetricsLexer
+	series         []byte
+	text           []byte
+	mtype          MetricType
+	val            float64
+	ts             int64
+	hasTS          bool
+	start          int
+	offsets        []int
+	prev           token
+	ignoreEofError bool // only used to simplify tests.
```

Comment:
No need to do this, the tests already allow for this.

Reply:
Without it all the tests errors are "data does not end with # EOF"

Reply:
I have updated the pull request to use the public methods only

Reply:
Update the testcases then

Reply:
Updated

Reply:
Now it looks close to the way we test the Python parser